phase1-coordinator/src/tests.rs

```diff
@@ -1244,6 +1254,219 @@ fn coordinator_drop_contributor_and_release_locks() {
+#[ignore]
+fn coordinator_drop_several_contributors() {
+    let parameters = Parameters::Custom(Settings {
+        contribution_mode: ContributionMode::Chunked,
+        proving_system: ProvingSystem::Groth16,
+        curve: CurveKind::Bls12_377,
+        power: 2,
+        batch_size: 2,
+        chunk_size: 2,
+    });
+    let replacement_contributor_1 = create_contributor_test_details("replacement-1");
+    let replacement_contributor_2 = create_contributor_test_details("replacement-2");
+    let testing = Testing::from(parameters).coordinator_contributors(&[
+        replacement_contributor_1.participant.clone(),
+        replacement_contributor_2.participant.clone(),
+    ]);
+    let environment = initialize_test_environment(&testing.into());
+
+    // Instantiate a coordinator.
+    let coordinator = Coordinator::new(environment, Box::new(Dummy)).unwrap();
+
+    // Initialize the ceremony to round 0.
+    coordinator.initialize().unwrap();
+    assert_eq!(0, coordinator.current_round_height().unwrap());
+
+    // Add some contributors and one verifier to the queue.
+    let contributor_1 = create_contributor_test_details("1");
+    let contributor_2 = create_contributor_test_details("2");
+    let contributor_3 = create_contributor_test_details("3");
+    let verifier_1 = create_verifier_test_details("1");
+    coordinator.add_to_queue(contributor_1.participant.clone(), 10).unwrap();
+    coordinator.add_to_queue(contributor_2.participant.clone(), 10).unwrap();
+    coordinator.add_to_queue(contributor_3.participant.clone(), 10).unwrap();
+    coordinator.add_to_queue(verifier_1.participant.clone(), 10).unwrap();
+
+    // Update the ceremony to round 1.
+    coordinator.update().unwrap();
+    assert_eq!(1, coordinator.current_round_height().unwrap());
+
+    // Make some contributions
+    let k = 3;
+    for _ in 0..k {
+        contributor_1.contribute_to(&coordinator).unwrap();
+        contributor_2.contribute_to(&coordinator).unwrap();
+        contributor_3.contribute_to(&coordinator).unwrap();
+
+        verifier_1.verify(&coordinator).unwrap();
+        verifier_1.verify(&coordinator).unwrap();
+        verifier_1.verify(&coordinator).unwrap();
+    }
+
+    let storage_lock = coordinator.storage();
+
+    {
+        let round = coordinator.current_round().unwrap();
+        let storage_read = storage_lock.read().unwrap();
+        let storage = &*storage_read;
+        check_round_matches_storage_files(&**storage, &round);
+    }
+
+    let _locators = coordinator.drop_participant(&contributor_1.participant).unwrap();
+    let _locators = coordinator.drop_participant(&contributor_2.participant).unwrap();
+
+    coordinator.update().unwrap();
+
+    {
+        let round = coordinator.current_round().unwrap();
+        let storage_read = storage_lock.read().unwrap();
+        let storage = &*storage_read;
+        check_round_matches_storage_files(&**storage, &round);
+    }
+
+    fn contribute_verify_until_no_tasks(
```

Comment:
Do you think this function is specific to the current test or may be reused between different tests?

Reply:
It could definitely be re-used in other tests when you don't care exactly how many contributions are required, you just want the thing to finish its job.